frontend/src/store/chats/actions.ts

```diff
@@ -0,0 +1,66 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import {
+  AsyncThunkConfig,
+  ChatMessageCreateRequestBodyDto,
+  ChatMessageGetAllItemResponseDto,
+  ChatMessageGetAllRequestParamsDto,
+  ChatMessageGetAllResponseDto,
+  ChatMessagesGetAllWithParticipantsDto,
+} from 'common/types/types';
+
+import { ActionType } from './common';
+
+const getLastMessages = createAsyncThunk<
+  ChatMessageGetAllResponseDto,
+  void,
+  AsyncThunkConfig
+>(ActionType.GET_LAST_MESSAGES, async (_, { extra }) => {
+  const { chatsApi } = extra;
+  const lastMessages = await chatsApi.getAllChatsLastMessages();
+
+  return lastMessages;
+});
+
+const getMessages = createAsyncThunk<
+  ChatMessagesGetAllWithParticipantsDto,
+  ChatMessageGetAllRequestParamsDto,
+  AsyncThunkConfig
+>(ActionType.GET_MESSAGES, async (payload, { extra }) => {
+  const { chatsApi } = extra;
+  const { id } = payload;
+
+  const messages = await chatsApi.getAllChatMessages(id);
+  let participants = null;
+
+  const [currentChatMessage] = messages.items;
+
+  if (currentChatMessage) {
+    participants = {
+      first: currentChatMessage.sender,
+      second: currentChatMessage.receiver,
+    };
+  }
```

Comment:
One of the participants is always the current user, you can just save `chatOpponentId`, then you'll not need ternary operator in `chats.tsx`

Reply:
good point